src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceService.java

```diff
@@ -77,35 +56,64 @@ public void updateAttendance(long id, UpdateAttendanceDto attendanceDto) throws
                         .createDateTime(attendanceData.getCreateDateTime())
                         .caseNoteId(attendanceData.getCaseNoteId())
                         .build())
-                  .collect(Collectors.toSet());
+                .collect(Collectors.toSet());
     }
 
     public AbsentReasonsDto getAbsenceReasons() {
         return new AbsentReasonsDto(AbsentReason.getPaidReasons(), AbsentReason.getUnpaidReasons());
     }
 
+    @Transactional
+    public void createAttendance(final CreateAttendanceDto attendanceDto) {
 
-    private void applyAttendanceWorkflow(String offenderNo, Attendance attendance) {
-         final var eventOutcome = nomisEventOutcomeMapper.getEventOutcome(
-                 attendance.getAbsentReason(),
-                 attendance.isAttended(),
-                 attendance.isPaid());
+        var attendance = attendanceRepository.save(toAttendance(attendanceDto));
+        entityManager.flush();
+        applyAttendanceWorkflow(attendance);
+    }
 
-         nomisService.putAttendance(offenderNo, attendance.getEventId(), eventOutcome);
+    @Transactional
+    public void updateAttendance(long id, UpdateAttendanceDto attendanceDto) throws AttendanceNotFound {
 
-         if (attendance.getAbsentReason() != null && AbsentReason.getIepTriggers().contains(attendance.getAbsentReason())) {
-             final var caseNote = nomisService.postCaseNote(
-                     attendance.getOffenderBookingId(),
-                     "NEG",//"Negative Behaviour"
-                     "IEP_WARN", //"IEP Warning",
-                     attendance.getComments(),
-                     LocalDateTime.now());
+        final var attendance = attendanceRepository.findById(id)
+                .orElseThrow(AttendanceNotFound::new);
 
-             attendance.setCaseNoteId(caseNote.getCaseNoteId());
-         }
-     }
+        attendance.setAttended(attendanceDto.getAttended());
+        attendance.setPaid(attendanceDto.getPaid());
+        attendance.setAbsentReason(attendanceDto.getAbsentReason());
+        attendance.setComments(attendanceDto.getComments());
+
+        applyAttendanceWorkflow(attendance);
+
+        attendanceRepository.save(attendance);
+    }
+
+
+    private void applyAttendanceWorkflow(Attendance attendance) {
+        final var eventOutcome = nomisEventOutcomeMapper.getEventOutcome(
+                attendance.getAbsentReason(),
+                attendance.getAttended(),
+                attendance.getPaid());
+
+        nomisService.putAttendance(attendance.getOffenderBookingId(), attendance.getEventId(), eventOutcome);
+
+        addCaseNoteIfRequired(attendance).ifPresent(attendance::setCaseNoteId);
+    }
+
+     private Optional<Long> addCaseNoteIfRequired(Attendance attendance) {
```

Comment:
Doesn't seem like you need to return anything here? the return value is not being used?

Reply:
The case note id is used on line 99.  I suspect I've misunderstood you though.

Reply:
oh no - I must have just misread..